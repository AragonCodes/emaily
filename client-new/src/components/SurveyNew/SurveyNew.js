// SurveyNew shows SurveyForm and SurveyFormReview
import React, { useState } from 'react';
import { reduxForm } from 'redux-form';
import SurveyForm from './common/SurveyForm';
import SurveyFormReview from './common/SurveyFormReview';

function SurveyNew() {
	const [reviewMode, setReviewMode] = useState(false);

	function renderContent() {
		return reviewMode
			? <SurveyFormReview onCancel={() => setReviewMode(false)} />
			: <SurveyForm onSurveySubmit={() => setReviewMode(true)} />
	}

	return (
		<div>
			{renderContent()}
		</div>
	);
}

export default reduxForm({
	form: 'surveyForm'
})(SurveyNew);
