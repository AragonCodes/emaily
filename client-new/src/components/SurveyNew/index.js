import SurveyNewComponent from './SurveyNew';

export default SurveyNewComponent;