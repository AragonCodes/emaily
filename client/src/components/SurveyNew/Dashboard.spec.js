import React from 'react';
import { shallow } from 'enzyme';
import SurveyNew from './SurveyNew';

describe('SurveyNew', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<SurveyNew />
		);
		expect(wrapper.debug()).toMatchSnapshot();
	});
});