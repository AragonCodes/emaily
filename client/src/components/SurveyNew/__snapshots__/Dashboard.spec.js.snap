// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyNew should render correctly 1`] = `"<Hoc touchOnBlur={true} touchOnChange={false} persistentSubmitErrors={false} destroyOnUnmount={true} shouldAsyncValidate={[Function: defaultShouldAsyncValidate]} shouldValidate={[Function: defaultShouldValidate]} shouldError={[Function: defaultShouldError]} shouldWarn={[Function: defaultShouldWarn]} enableReinitialize={false} keepDirtyOnReinitialize={false} updateUnregisteredFields={false} getFormState={[Function: getFormState]} pure={true} forceUnregisterOnUnmount={false} submitAsSideEffect={false} form=\\"surveyForm\\" forwardedRef={{...}} />"`;
